redis/src/main/java/org/togglz/redis/RedisStateRepository.java

```diff
@@ -0,0 +1,151 @@
+package org.togglz.redis;
+
+import org.togglz.core.Feature;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.core.repository.StateRepository;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.Protocol;
+
+import java.util.Map;
+
+/**
+ * A state repository which stores the feature state in Redis.
+ * <p>
+ * The class provides a builder which can be used to configure the state repository instance
+ * (e.g. Redis hostname, Redis/Jedis configuration, serialization).
+ * </p>
+ *
+ * @author Cosmin Rentea
+ */
+public class RedisStateRepository implements StateRepository {
+
+    public static final String KEY_PREFIX = "togglz:";
+    public static final String ENABLED_FIELD = "enabled";
+    public static final String STRATEGY_FIELD = "strategy";
+    public static final String PARAMETER_PREFIX = "parameter:";
+    public static final int PARAMETER_PREFIX_LENGTH = PARAMETER_PREFIX.length();
+
+    protected final JedisPool jedisPool;
```

Comment:
I added another method to the builder - taking a JedisPool as a parameter. 
If no JedisPool is explicitly set, then the JedisPoolConfig + hostname will be used.
Removed  also the "jedisPoolConfig" from the fields of the RedisStateRepository class, since only "jedisPool" is actually used after the initialization.
Please review the changes.
Thank you for the feedback!